File: src/list.js
import chalk from "chalk";
import { loadPackages } from "../utils";

export const list = async () => {
  const db = await loadPackages();

  if (Object.keys(db).length === 0) {
    console.log('No packages installed');
  }
  for (const [k, v] of Object.entries(db)) {
    console.log(chalk.bold(k), v.map(e => e.enabled == true ? `${chalk.bold(e.version)} (enabled)` : e.version).reverse().join(', '));
  }
};

File: lib/commands/list.dart
import 'package:collection/collection.dart';
import 'package:tint/tint.dart';
import 'package:zapstore_cli/models/package.dart';

void list([String? filter]) async {
  final db = await Package.loadAll();

  if (db.isEmpty) {
    print('No packages installed');
  }

  final regexp = filter != null ? RegExp(filter) : null;

  final orderedEntries = db.entries
      .where((e) => regexp?.hasMatch(e.key) ?? true)
      .sortedBy((e) => e.key);
  for (final MapEntry(:key, value: package) in orderedEntries) {
    print('${key.bold()}: ${package.version}');
  }
}
